src/main/java/org/andersen/homework/util/collection/MyArrayList.java

```diff
@@ -0,0 +1,80 @@
+package org.andersen.homework.util.collection;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class MyArrayList<E> {
+
+  private static final Object[] DEFAULT_EMPTY_CAPACITY_ARRAY = {};
+
+  private Object[] array;
+
+  private int capacity;
+
+  private int entriesCount = 0;
+
+  public MyArrayList() {
+    this.array = DEFAULT_EMPTY_CAPACITY_ARRAY;
+    this.capacity = 0;
+  }
+
+  public MyArrayList(int initialCapacity) {
+    if (initialCapacity > 0) {
+      this.array = new Object[initialCapacity];
+    } else if (initialCapacity == 0) {
+      this.array = DEFAULT_EMPTY_CAPACITY_ARRAY;
+    } else {
+      throw new IllegalArgumentException("Illegal Capacity: " + initialCapacity);
+    }
+
+    this.capacity = initialCapacity;
+  }
+
+  @SuppressWarnings("unchecked")
+  public E get(int index) {
+    Objects.checkIndex(index, entriesCount);
```

Comment:
What does this method do?)

Reply:
Checks if 'index' is out of bounds, throws exception if so